content/blog/exposing-an-application-using-ingress-and-tls-termination-on-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,528 @@
+---
+title: Exposing applications using Ingress and TLS termination on Kubernetes in
+  HPE GreenLake for Private Cloud Enterprise
+date: 2024-03-20T10:06:33.740Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+disable: false
+tags:
+  - HPE GreenLake for Private Cloud Enterprise
+  - Kubernetes
+  - Ingress
+  - Ingress Controller
+  - TLS Termination
+  - Load balancer
+  - metalLB
+  - cert-manager
+  - Nginx Ingress controller
+  - SSL/TLS certificates
+  - hpe-greenlake-for-private-cloud-enterprise
+---
+<style> li { font-size: 27px; line-height: 33px; max-width: none; } </style>
+
+### Overview
+
+[HPE GreenLake for Private Cloud Enterprise: Containers](https://www.hpe.com/us/en/greenlake/containers.html), one of the HPE GreenLake cloud services available on the HPE GreenLake for Private Cloud Enterprise, allows customers to create a K8s cluster and deploy containerized applications to the cluster. It provides an enterprise-grade container management service using open source K8s.  
+
+O﻿nce applications are deployed in a K8s cluster, the next step is to create services that expose these applications. By default, K8s services are created with the *ClusterIP* type, which supports internal connectivity among different components of the applications. However, these services are not directly accessible from outside the cluster. The challenge arises with a requirement to securely expose the deployed applications over HTTPS. This involves generating and managing SSL/TLS certificates for multiple applications deployed in the cluster. These certificates are crucial for secure communication between services, and their correct installation and management are essential to avoid access issues and security risks. To address exposing applications over HTTPS, K8s provides the concept of *Ingress*. An Ingress acts as an entry point for external traffic into the cluster. It can be configured with TLS termination. However, setting up K8s Ingress with TLS termination is intricate. It involves creating a K8s Secret to host the certificate and referencing the Secret in the Ingress resource. It also requires an additional load balancer configuration in the cluster.
+
+This blog post outlines the comprehensive steps for exposing applications u﻿sing Ingress and TLS termination on K8s in HPE GreenLake for Private Cloud Enterprise. [MetalLB](https://metallb.universe.tf/) is deployed to the cluster to set up the load balancer. It enables external access to services within the cluster. [Cert-manager](https://cert-manager.io/) is utilized for creating and managing SSL/TLS certificates. The generated certificate is stored as a K8s *Secret* object. This Secret can be mounted by application Pods or used by an Ingress controller. The [Nginx Ingress controller](https://www.nginx.com/products/nginx-ingress-controller/) is deployed and configured in the cluster. It handles SSL certificates and facilitates secure access to applications in the backend.
```

Comment:
Good point. I was thinking about this when I was creating the context. The same thing will apply for MetalLB. Instead of adding references to MetalLB & cert-manager, I can replace them with 2 blog posts. What do you think ?

Reply:
Yes, I believe it would be a very good idea to make a cross reference to your blog post about MetaLB (https://developer.hpe.com/blog/set-up-load-balancer-with-metallb-in-hpe-greenlake-for-private-cloud-enterprise/).